fieldmask/update.go

```diff
@@ -47,11 +49,18 @@ func Update(mask *fieldmaskpb.FieldMask, dst, src proto.Message) {
 
 func updateWireSetFields(dst, src protoreflect.Message) {
 	src.Range(func(field protoreflect.FieldDescriptor, value protoreflect.Value) bool {
-		if isMessage(field) {
+		switch {
+		case field.IsList():
+			dst.Set(field, value)
+		case field.IsMap():
+			dst.Set(field, value)
+		case field.Message() != nil && !dst.Has(field):
```

Comment:
Oh snap, `protoreflect.Message.Has` is the method we have been looking for in ValidateRequiredFields, I think!

Reply:
🎉